alicloud/service_alicloud_rds.go

```diff
@@ -813,6 +822,33 @@ func (s *RdsService) WaitForAccountPrivilegeRevoked(id, dbName string, timeout i
 	return nil
 }
 
+func (s *RdsService) WaitForDBDatabase(id string, status Status, timeout int) error {
+	parts, err := ParseResourceId(id, 2)
+	if err != nil {
+		return WrapError(err)
+	}
+	deadline := time.Now().Add(time.Duration(timeout) * time.Second)
+	for {
+		object, err := s.DescribeDBDatabase(id)
+		if err != nil {
+			if NotFoundError(err) {
+				if status == Deleted {
+					return nil
+				}
+			}
+			return WrapError(err)
+		}
+		if object.DBName == parts[1] {
+			break
+		}
+		time.Sleep(DefaultIntervalShort * time.Second)
+		if time.Now().After(deadline) {
+			return WrapErrorf(err, WaitTimeoutMsg, id, GetFunc(1), timeout, "", parts[1], ProviderERROR)
```

Comment:
The line 846 should be `return WrapErrorf(err, WaitTimeoutMsg, id, GetFunc(1), timeout, object.DBName, parts[1], ProviderERROR)`